src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -433,13 +585,50 @@ protected virtual void Dispose(bool disposing)
         /// <returns>A task whose result is the deserialized response from the JSON-RPC server.</returns>
         protected virtual async Task<ReturnType> InvokeCoreAsync<ReturnType>(int? id, string targetName, IReadOnlyList<object> arguments, CancellationToken cancellationToken)
         {
+            return await InvokeCoreAsync<ReturnType>(id, targetName, arguments, cancellationToken, isParameterObject: false).ConfigureAwait(false);
```

Comment:
optimization tip: A thunk method like this doesn't need to be async. If you remove the `async` keyword from the method signature, the `await` keyword from this expression, and the `.ConfigureAwait(false)` from the end of the expression, you'll eliminate the state machine that the compiler and runtime will have to create for an async method.